File: examples/integration/annotations/band.py
from __future__ import absolute_import

from bokeh.io import save
from bokeh.models import Plot, Range1d, Band, ColumnDataSource

x_range = Range1d(0, 10)
y_range = Range1d(0, 10)

# Have to specify x/y range as labels aren't included in the plot area solver
plot = Plot(plot_width=600, plot_height=600,
            x_range=x_range, y_range=y_range, toolbar_location=None)

source = ColumnDataSource(data=dict(
    x1 = [1,3,5,7,9],
    lower1 = [1,2,1,2,1],
    upper1 = [2,3,2,3,2],
    x2 = [200, 250, 350, 450, 550],
    lower2 = [400, 300, 400, 300, 400],
    upper2 = [500, 400, 500, 400, 500],
))

band1 = Band(base='x1', lower='lower1', upper='upper1',
             line_width=3, line_color='red', line_dash='dashed',
             source=source)
band2 = Band(base='x2', lower='lower2', upper='upper2',
             base_units='screen', lower_units='screen', upper_units='screen',
             dimension='width', line_width=3, fill_color='blue', line_color='green',
             source=source)

plot.add_layout(band1)
plot.add_layout(band2)

save(plot)

File: bokehjs/test/integration/annotations/band.ts
import {display, fig} from "../_util"

import {Band, ColumnDataSource} from "@bokehjs/models"

describe("Band annotation", () => {

  it("should support basic positioning", async () => {
    const plot = fig([300, 300], {x_range: [0, 10], y_range: [0, 10]})

    const source = new ColumnDataSource({
      data: {
        x1: [1, 3, 5, 7, 9],
        lower1: [1, 2, 1, 2, 1],
        upper1: [2, 3, 2, 3, 2],
        x2: [50, 100, 150, 200, 250],
        lower2: [100, 50, 100, 50, 100],
        upper2: [200, 150, 200, 150, 200],
      },
    })

    const band0 = new Band({
      base: {field: "x1"},
      lower: {field: "lower1"},
      upper: {field: "upper1"},
      line_width: 3, line_color: "red", line_dash: "dashed",
      source,
    })
    const band1 = new Band({
      // TODO: units are only supported on value level, not type level
      base: {field: "x2", units: "screen"} as any,
      lower: {field: "lower2", units: "screen"} as any,
      upper: {field: "upper2", units: "screen"} as any,
      dimension: "width", line_width: 3, fill_color: "blue", line_color: "green",
      source,
    })

    plot.add_layout(band0)
    plot.add_layout(band1)

    await display(plot)
  })
})
